ios-apps-master/Beer_Diary/Beer_Diary/Add_Beer.swift

```diff
@@ -0,0 +1,256 @@
+    @IBOutlet weak var aromaSlider: UISlider!
+    @IBOutlet weak var appearanceSlider: UISlider!
+    @IBOutlet weak var palateSlider: UISlider!
+    @IBOutlet weak var tasteSlider: UISlider!
+    
+    
+    var newItem = BeerItem()
+    
+    
+    
+    
+    @IBAction func sliderAroma(_ sender: Any)
+    {
+        let tempVal = Int(aromaSlider.value)
+        aromaSlider.value = Float(tempVal)
+        newItem.aroma = tempVal
+        aroma.text = "Aroma - \(tempVal)/10"
+        newItem.overall = Float(Int(Float(newItem.aroma + newItem.appearance + newItem.palate + newItem.taste)/30*500))/100
+        overall.text = "Overall - \((newItem.overall))/5"
+    }
+    
+    @IBAction func sliderAppearance(_ sender: Any)
+    {
+        let tempVal = Int(appearanceSlider.value)
+        appearanceSlider.value = Float(tempVal)
+        newItem.appearance = tempVal
+        appearance.text = "Appearance - \(tempVal)/5"
+        newItem.overall = Float(Int(Float(newItem.aroma + newItem.appearance + newItem.palate + newItem.taste)/30*500))/100
+        overall.text = "Overall - \((newItem.overall))/5"
+    }
+    
+    @IBAction func sliderPalate(_ sender: Any)
+    {
+        let tempVal = Int(palateSlider.value)
+        palateSlider.value = Float(tempVal)
+        newItem.palate = tempVal
+        palate.text = "Palate - \(tempVal)/5"
+        newItem.overall = Float(Int(Float(newItem.aroma + newItem.appearance + newItem.palate + newItem.taste)/30*500))/100
+        overall.text = "Overall - \((newItem.overall))/5"
+    }
+    
+    @IBAction func sliderTaste(_ sender: Any)
+    {
+        let tempVal = Int(tasteSlider.value)
+        tasteSlider.value = Float(tempVal)
+        newItem.taste = tempVal
+        taste.text = "Taste - \(tempVal)/10"
+        newItem.overall = Float(Int(Float(newItem.aroma + newItem.appearance + newItem.palate + newItem.taste)/30*500))/100
+        overall.text = "Overall - \((newItem.overall))/5"
+    }
+    
+    
+    @IBAction func add(_ sender: Any)
+    {
+        if let string = beerName.text
+        {
+            newItem.name = string
+        }
+        if let string = breweryName.text
+        {
+            newItem.brewery = string
+        }
+        if let string = styleName.text
+        {
+            newItem.style = string
+        }
+        if let string = countryName.text
+        {
+            newItem.country = string
+        }
+        if let string = thoughts.text
+        {
+            newItem.thoughts = string
+        }
+        
+        
+        
+       if (newItem.aroma != 0) && (newItem.appearance != 0) && (newItem.palate != 0) && (newItem.taste != 0) && (newItem.name != "") && (newItem.brewery != "") && (newItem.style != "") && (newItem.country != "") && (newItem.thoughts != "")
+       {
+            newItem.date = NSDate()
+        
+            BeerList.append(newItem)
+        
+            //UserDefaults.standard.set(BeerList, forKey: "BeerList")
```

Comment:
Don't leave commented in code, it makes it all messy, better to remove and keep old stuff in git